lib/semian.rb

```diff
@@ -124,7 +124,8 @@ def register(name, tickets:, permissions: 0660, timeout: 0, error_threshold:, er
       error_threshold: error_threshold,
       error_timeout: error_timeout,
       exceptions: Array(exceptions) + [::Semian::BaseError],
-      implementation: ::Semian::Simple,
+      permissions: permissions,
+      implementation: ::Semian::SysV,
```

Comment:
How I did this before was that the `SysV::` classes can themselves function as the fallback, since they inherit from the `Simple::` classes. Distinguishing between the two states was the reason I had `@using_shared_memory` as a variable.
I guess what you're imagining is more of a 
```ruby
implementation: semaphores_enabled? ? ::Semian::SysV : ::Semian::Simple
```